cfme/storage/volume.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+from navmazing import NavigateToSibling, NavigateToAttribute
+from widgetastic.widget import View, Text, NoSuchElementException
+from widgetastic_patternfly import Button, Dropdown, FlashMessages
+from widgetastic_manageiq import (
+    ItemsToolBarViewSelector, Search, Table, PaginationPane, BreadCrumb, SummaryTable, Accordion,
+    ManageIQTree, TextInput, BootstrapSelect)
+
+from cfme.base.ui import BaseLoggedInPage
+from cfme.exceptions import VolumeNotFound
+from cfme.web_ui import match_location
+from utils.appliance import Navigatable
+from utils.appliance.implementations.ui import CFMENavigateStep, navigator
+from utils.log import logger
+
+
+class VolumeToolbar(View):
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Dropdown('Download')  # title match
+    view_selector = View.nested(ItemsToolBarViewSelector)
+
+
+class VolumeDetailsToolbar(View):
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Button('Download summary in PDF format')
+
+
+class VolumeEntities(View):
+    title = Text('//div[@id="main-content"]//h1')
+    table = Table("//div[@id='list_grid']//table")
+    search = View.nested(Search)
+    flash = FlashMessages('.//div[starts-with(@id, "flash_text_div")]')
+
+
+class VolumeDetailsEntities(View):
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+    properties = SummaryTable('Properties')
+    relationships = SummaryTable('Relationships')
+    smart_management = SummaryTable('Smart Management')
+
+
+class VolumeDetailsAccordion(View):
+    @View.nested
+    class properties(Accordion):  # noqa
+        tree = ManageIQTree()
+
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
+
+
+class VolumeView(BaseLoggedInPage):
+    """Base class for header and nav check"""
+    @property
+    def in_volume(self):
+        nav = ['Storage', 'Volumes'] if self.context['object'].appliance.version < '5.8' else \
+            ['Storage', 'Block Storage', 'Volumes']
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == nav and
+            match_location(controller='cloud_volume', title='Cloud Volumes'))
+
+
+class VolumeAllView(VolumeView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_volume and
+            self.entities.title.text == 'Cloud Volumes')
+
+    toolbar = View.nested(VolumeToolbar)
+    entities = View.nested(VolumeEntities)
+    paginator = View.nested(PaginationPane)
```

Comment:
I think you can replace the View.nested with `Class()` here, but depends on you. The semantics is pretty much the same: https://github.com/RedHatQE/widgetastic.core/blob/master/src/widgetastic/widget.py#L589 (constructing an instance without browser or parent widget as first arg yields the descriptor anyway)